### PR TITLE
feat(python): add task celery to prune events logs

### PR DESCRIPTION
**Pull request by jplanckeel**
### SUMMARY
After exchange with superset contributor we find add a celery task a good idea to not impact users.

Add a task celery schedul one time per day to prune events log in tables logs. 

This PR is a reponse of this exchange : https://github.com/apache/superset/pull/20636 

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
image

### ADDITIONAL INFORMATION

- [x] Has associated issue: https://github.com/apache/superset/issues/16528
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [x] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/utils/log.py [hunk 1]
```diff
@@ -352,3 +355,71 @@ def log(  # pylint: disable=too-many-arguments,too-many-locals
         except SQLAlchemyError as ex:
             logging.error("DBEventLogger failed to log event(s)")
             logging.exception(ex)
+
+
+class AsyncPruneEventScheduleLogCommand(AbstractEventLogger):
```
Comment (craig-rueda) [inline superset/utils/log.py:360]: Pls put this under the `superset.commands.logs` package

File: superset/utils/log.py [hunk 2]
```diff
@@ -352,3 +355,71 @@ def log(  # pylint: disable=too-many-arguments,too-many-locals
...
+                except DAODeleteFailedError as ex:
+                    prune_errors.append(str(ex))
+            if prune_errors:
+                raise "An error occurred while pruning events logs;".join(prune_errors)
+
+    def validate(self) -> None:
+        pass
+
+
+def bulk_delete_logs(
```
Comment (craig-rueda) [inline superset/utils/log.py:398]: Move this method into the `Command` It's not really pure as it relies on `db.session`, which is a proxy

File: superset/config.py [hunk 3]
```diff
@@ -73,6 +73,7 @@
 # Realtime stats logger, a StatsD implementation exists
 STATS_LOGGER = DummyStatsLogger()
 EVENT_LOGGER = DBEventLogger()
+EVENT_LOG_PRUNE = 90
```
Comment (craig-rueda) [inline superset/config.py:76]: Pls add comments around this config prop

**Review by craig-rueda — CHANGES_REQUESTED** [review]
Thanks for this addition! Overall, looks good except for a few comments.

Also, please add tests for your new command / task

File: superset/utils/log.py [hunk 4]
```diff
@@ -352,3 +355,71 @@ def log(  # pylint: disable=too-many-arguments,too-many-locals
...
+        
+        with session_scope(nullpool=True) as session:
+            self.validate()
+            prune_errors = []
+
+            if log_retention is not None:
+                from_date = datetime.utcnow() - timedelta(days=log_retention)
+                try:
+                    row_count = bulk_delete_logs(
+                        from_date, session=session, commit=False
```
Comment (mayurnewase) [inline superset/utils/log.py:383]: no commit?

**Comment by mayurnewase** [comment]
need to use abstraction for commands + unit tests.

File: superset/utils/log.py (hunk 2 above)
Comment (dpgaspar) [reply superset/utils/log.py:398]: He is following the same pattern that already exists, I'de vote for keeping it here

**Comment by NickLarsenNZ** [comment]
This PR is very old. Is it still worth having?
@jplanckeel @craig-rueda

**Comment by rusackas** [comment]
Agreed, @NickLarsenNZ.

@craig-rueda @mayurnewase would either of you want to adapt/adopt this work in a new PR? Maybe we can just close this if not.

**Comment by craig-rueda** [comment]
Closing for now.

**Pull request closed without merging**